File: src/main/java/org/processmining/models/Movable.java
package org.processmining.models;

import org.deckfour.xes.model.XTrace;

/**
 * Created by Ivan Shugurov on 22.10.2014.
 */
public interface Movable
{
    MovementResult move(XTrace trace);

    boolean checkAvailability();
}

File: src/main/java/org/processmining/models/Movable.kt
package org.processmining.models

import org.deckfour.xes.model.XTrace

/**
 * Created by Ivan Shugurov on 22.10.2014.
 */
interface Movable {
    fun move(trace: XTrace): MovementResult<*>?
    
    fun checkAvailability(): Boolean
}
